### PR TITLE
Api updates and fixes

### LINKED ISSUE #118: State root mismatch post sync
It has been observed that post sync the state root obtained is not the same as post apply post validation. 

This is likely related to #116

### PR DESCRIPTION
**Pull request by vaibhavchellani**
Fixes #118 #116

### CODE REVIEW

File: cmd/rest.go [hunk 1]
```diff
@@ -19,13 +20,15 @@ func startRestServerCmd() *cobra.Command {
 			cfg, err := config.ParseConfig()
 			common.PanicIfError(err)
 			r, err := rest.LoadRouters(cfg)
+			originsOk := handlers.AllowedOrigins([]string{"*"})
```
Comment (ChihChengLiang) [inline cmd/rest.go:23]: Why we are doing CORS in the coordinator client? What's the use case here? Usually, this should be handled in a gateway nginx or something in the server that the coordinator client is deployed.

Comment (vaibhavchellani) [reply cmd/rest.go:23]: Yeah the next step would be allowing to configure these properties in the `config.toml` so a user has options to run commander by itself or behind an API gateway for eg: nginx

Comment (vaibhavchellani) [reply cmd/rest.go:23]: It just seems right to allow users to configure the HTTP server itself with the additional option of putting it behind a API gateway and configuring that.

Open to changing this.

File: rest/handler.go [hunk 2]
```diff
@@ -267,6 +288,105 @@ func create2transferTx(w http.ResponseWriter, r *http.Request) {
...
+		if err != nil {
+			WriteErrorResponse(w, http.StatusBadRequest, "Unable to decode state")
+			return
+		}
+		var stateData UserDetailsState
+		stateData.Balance = balance.Uint64()
+		stateData.Nonce = nonce.Uint64()
+		stateID, err := core.StringToUint(state.Path)
+		if err != nil {
+			fmt.Println("error converting path to ID")
```
Comment (ChihChengLiang) [inline rest/handler.go:370]: should use log and log the path that failed. so that the debug is easier.

**Review by ChihChengLiang — APPROVED** [review]
LGTM

File: cmd/rest.go (hunk 1 above)
Comment (ChihChengLiang) [reply cmd/rest.go:23]: Per our offline discussion, let's add cors to unblock wallet development and fix it later. #120 is the issue for this.

File: rest/handler.go (hunk 2 above)
Comment (vaibhavchellani) [reply rest/handler.go:370]: yeah let me handle all errors in another PR

**Pull request merged**